src/Contacts/src/FixedFootDetector.cpp

```diff
@@ -28,45 +35,54 @@ bool FixedFootDetector::customInitialization(std::weak_ptr<const IParametersHand
 
     if (!ptr->getParameter("sampling_time", m_dT))
     {
-        log()->error("{} Unable to find the sampling time.", logPrefix);
+        log()->error("{} Unable to find the 'sampling_time' parameter.", logPrefix);
         return false;
     }
 
     if (m_dT <= 0)
     {
-        log()->error("{} The sampling time must be a strictly positive number.", logPrefix);
+        log()->error("{} The parameter 'sampling_time' must be a strictly positive number.",
+                     logPrefix);
         return false;
     }
 
+    m_detectorState = State::Initialized;
+
     return true;
 }
 
 bool FixedFootDetector::updateFixedFoot()
 {
-    assert(m_currentTime >= m_contactPhaselist.firstPhase()->beginTime
-           && "[FixedFootDetector::updateFixedFoot] The current time must be greater than "
-              "equal the first contact phase. If you read this assert there is a bug in the code");
+    constexpr auto logPrefix = "[FixedFootDetector::updateFixedFoot]";
 
     // search the phase associated to the current time
-    auto phase = std::find_if(m_contactPhaselist.cbegin(),
-                              m_contactPhaselist.cend(),
-                              [&](const auto& phase) -> bool {
-                                  return phase.beginTime <= m_currentTime
-                                      && m_currentTime <= phase.endTime;
-                              });
+    constexpr auto tolerance = std::numeric_limits<double>::min();
+    auto phase = m_contactPhaselist.getPresentPhase(m_currentTime, tolerance);
```

Comment:
We might also consider ``std::numeric_limits<double>::min();`` as default tolerance

Reply:
Done in https://github.com/ami-iit/bipedal-locomotion-framework/commit/9c1a49364634ca8d52439d19e444ce13c30efb85